cmd/update.go

```diff
@@ -0,0 +1,69 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ONSdigital/dp-sitemap/cmd/utilities"
+	"github.com/ONSdigital/dp-sitemap/config"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// updateCmd represents the update command
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Generate sitemap files using default parameters",
+	Long:  `A tool to update generate and update sitemap files`,
+	Run: func(cmd *cobra.Command, args []string) {
+		cfg, err := config.Get()
+		if err != nil {
+			fmt.Println("Error retrieving config"+err.Error(), cfg)
+			os.Exit(1)
+		}
+		viper.BindPFlags(cmd.Flags()) // Bind Flags with Viper
+
+		// Validate ApiUrl
+		if !isValidUrl(viper.GetString("api-url")) {
+			fmt.Printf("api-url is not a valid URL: %s\n", viper.GetString("api-url"))
+			os.Exit(1)
+		}
+
+		// Validate ZebedeeUrl
+		if !isValidUrl(viper.GetString("zebedee-url")) {
+			fmt.Printf("zebedee-url is not a valid URL: %s\n", viper.GetString("zebedee-url"))
+			os.Exit(1)
+		}
+
+		// Create FlagFields structure
+		flagList := utilities.FlagFields{
+			RobotsFilePath:  viper.GetString("robots-file-path"),
+			ApiUrl:          viper.GetString("api-url"),
+			SitemapIndex:    viper.GetString("sitemap-index"),
+			ScrollTimeout:   viper.GetString("scroll-timeout"),
+			ScrollSize:      viper.GetInt("scroll-size"),
+			SitemapPath:     viper.GetString("sitemap-file-path"),
+			ZebedeeUrl:      viper.GetString("zebedee-url"),
+			FakeScroll:      viper.GetBool("fake-scroll"),
+			GenerateSitemap: viper.GetBool("generate-sitemap"),
+			UpdateSitemap:   viper.GetBool("update-sitemap"),
```

Comment:
I don't think you need these now you've separated out the commands with cobra?

Reply:
This is the standard way to use cobra with viper. I have use viper to do a validation first and then assigned the validated flags into our FlagField structure.